flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamGraphGeneratorTest.java

```diff
@@ -460,13 +461,19 @@ public void testIteration() {
 		DataStream<Integer> filter = map.filter((x) -> false).name("filter").setParallelism(2);
 		iteration.closeWith(filter).print();
 
+		final ResourceSpec resources = ResourceSpec.newBuilder(1.0, 100).build();
+		iteration.getTransformation().setResources(resources, resources);
+
 		StreamGraph streamGraph = env.getStreamGraph();
 		for (Tuple2<StreamNode, StreamNode> iterationPair : streamGraph.getIterationSourceSinkPairs()) {
 			assertNotNull(iterationPair.f0.getCoLocationGroup());
 			assertEquals(iterationPair.f0.getCoLocationGroup(), iterationPair.f1.getCoLocationGroup());
 
 			assertEquals(StreamGraphGenerator.DEFAULT_SLOT_SHARING_GROUP, iterationPair.f0.getSlotSharingGroup());
 			assertEquals(iterationPair.f0.getSlotSharingGroup(), iterationPair.f1.getSlotSharingGroup());
+
+			assertEquals(resources, iterationPair.f0.getMinResources());
+			assertEquals(ResourceSpec.ZERO, iterationPair.f1.getMinResources());
```

Comment:
That's right. Testing the merged resources would be better since the way to split the resources should make no difference at the moment and we actually do not care about it.